src/NosCore.GameObject/Character.cs

```diff
@@ -716,36 +721,36 @@ public void Buy(Shop shop, short slot, short amount)
             }
         }
 
-        private NInvPacket BuyFrom(ShopItem item, short amount, short slotChar)
+        private async Task<NInvPacket?> BuyFrom(ShopItem item, short amount, short slotChar)
         {
             var type = item.Type;
-            var itemInstance = amount == item.ItemInstance.Amount
+            var itemInstance = amount == item.ItemInstance?.Amount
                 ? InventoryService.DeleteById(item.ItemInstance.Id)
-                : InventoryService.RemoveItemAmountFromInventory(amount, item.ItemInstance.Id);
+                : InventoryService.RemoveItemAmountFromInventory(amount, item.ItemInstance!.Id);
             var slot = item.Slot;
-            item.Amount -= amount;
+            item.Amount = (short)((item.Amount ?? 0) - amount);
             if ((item?.Amount ?? 0) == 0)
             {
-                Shop.ShopItems.TryRemove(slot, out _);
+                Shop!.ShopItems.TryRemove(slot, out _);
             }
 
-            SendPacket(itemInstance.GeneratePocketChange((PocketType)type, slotChar));
-            SendPacket(new SMemoPacket
+            await SendPacket(itemInstance.GeneratePocketChange((PocketType)type, slotChar));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5262615)